lib/rubocop/cop/mixin/hash_shorthand_syntax.rb

```diff
@@ -133,8 +133,16 @@ def use_modifier_form_without_parenthesized_method_call?(ancestor)
       end
 
       def last_expression?(ancestor)
-        ancestor.right_sibling ||
-          ancestor.each_ancestor.find { |node| node.assignment? || node.send_type? }&.right_sibling
+        return false if ancestor.right_sibling
```

Comment:
As the logic in the method became a bit dense I'd sparkle a few comments describing what exactly are we checking for in the various branches.

Reply:
I have simplified the method, I hope it's self explanatory now. The idea is that if you have a right node, there is code afterwards, unless the current node is the right node of an assignment. Then you have to check the right sibling of the assignment node itself. This can be recursive, because the current node can be a right sibling of an assignment itself, like the multiple assignment case I hit.

@koic there used to be a condition about being the right node of a `send` (method dispatch). Removing this does not fail any tests, are you sure the check was needed and the current approach is exhaustive?

Reply:
I would have written test as much as possible, so it may be removed I think. If the deletion caused an error, a test may have been forgotten.